<launch>
  <rosparam file="$(find roboticarm)/config/arm_control.yaml" command="load"/>

  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
  </include>

  <param name="robot_description" command="$(find xacro)/xacro $(find roboticarm)/urdf/robotic_arm.urdf" />

  <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-urdf -model joint_controller -param robot_description" />

  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="false" output="screen">
    <remap from="/joint_states" to="/joint_controller/joint_states"/>
  </node>

  <node 
    name="controller_spawner" 
    pkg="controller_manager" 
    type="spawner" 
    respawn="false"
    output="screen" 
    ns="/joint_controller" 
    args="joint1_position_controller joint2_position_controller joint3_position_controller joint4_position_controller joint_state_controller"/>
</launch>